File: Desain pattern/Mediator vs Observer/Observer/Enemy.cpp
#include "Enemy.h"

Enemy::Enemy(int hp) : health(hp) {}

void Enemy::takeDamage(int amount) {
    health -= amount;
    if (health < 0) health = 0;
}

bool Enemy::isDead() const {
    return health <= 0;
}

int Enemy::getHealth() const {
    return health;
}

File: Game Controler/Assets/Heavy Weapon/Script/Enemy.cs
using UnityEngine;

public class Enemy : MonoBehaviour
{
    public EnemyData enemyData;
    private float currentHealth;

    private void Start()
    {
        currentHealth = enemyData.maxHealth;
    }

    public void TakeDamage(float amount)
    {
        currentHealth -= amount;
        Debug.Log($"{gameObject.name} took {amount} damage! Remaining HP: {currentHealth}");

        if (currentHealth <= 0)
        {
            Die();
        }
    }

    private void Die()
    {
        Debug.Log($"{gameObject.name} died!");
        Destroy(gameObject);
    }
}
